### DISCUSSION TITLE
`map_err` vs `Ok(?)`

### DISCUSSION THREAD

**Discussion by meowjesty** [metalbear-co/mirrord#210] [Polls]
Which style should we use for functions that return `Result` (example from `mirrord-layer/src/file/ops.rs`):

```rust
// current style
file_channel_rx.blocking_recv()?.map_err(From::from)

// alternative style
Ok(file_channel_rx.blocking_recv()??)
```

The `map_err` will still be needed in some places (functions that do **not** return `Result` for example), but for most of the code both are valid.

**Comment by infiniteregrets** [comment]
Honestly, both look fine to me, if we are using map_err everywhere then we should probably stick to that otherwise the Ok() style looks more sleek

**Reply by eyalb181** [reply]
Agreed that Ok looks clearer (and our codebase is still small enough that we don't have to make decisions based on what's already being used).
